lib/core/src/client/preview/start.js

```diff
@@ -11,6 +12,50 @@ import reducer from './reducer';
 import * as Actions from './actions';
 import syncUrlWithStore from './syncUrlWithStore';
 
+const classes = {
+  MAIN: 'sb-show-main',
+  NOPREVIEW: 'sb-show-nopreview',
+  ERROR: 'sb-show-errordisplay',
+};
+
+function showMain() {
+  document.body.classList.remove(classes.NOPREVIEW);
+  document.body.classList.remove(classes.ERROR);
+
+  document.body.classList.add(classes.MAIN);
+}
+
+function showNopreview() {
+  document.body.classList.remove(classes.MAIN);
+  document.body.classList.remove(classes.ERROR);
+
+  document.body.classList.add(classes.NOPREVIEW);
+}
+
+function showErrorDisplay({ message, stack }) {
+  document.getElementById('error-message').textContent = message;
```

Comment:
Interesting, we're just using plain JS here to show/hide plain messages.

Makes sense.. I always assumed the error handling and debugging experiences for the different frameworks would be wildly different and thus would warrant having a separate component for each.

But setup is quite nice and simple I guess, description has to be an html-string I guess, and so a framework could submit some rich data down to the error display.

Reply:
> description has to be an html-string

Well right now it isn't supported (I use `textContent` not `innerHTML`). Error stack can be user-generated, so there's some room for xss here

Reply:
But they were enough so far

Reply:
What is XSS in this context?

Reply:
As in who is attacking whom?

Reply:
Somebody giving you some tricky URL when you have public storybook on some route of your main website whicj has auth and cookies

Reply:
Oooh, that's a good one, touche